File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/coupon/service/UserCouponService.java
package com.eager.questioncloud.core.domain.payment.coupon.service;

import com.eager.questioncloud.core.domain.payment.coupon.dto.UserCouponDto.AvailableUserCouponItem;
import com.eager.questioncloud.core.domain.payment.coupon.implement.UserCouponReader;
import com.eager.questioncloud.core.domain.payment.coupon.implement.UserCouponRegister;
import com.eager.questioncloud.core.domain.payment.coupon.model.UserCoupon;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserCouponService {
    private final UserCouponRegister userCouponRegister;
    private final UserCouponReader userCouponReader;

    public UserCoupon registerCoupon(Long userId, String couponCode) {
        return userCouponRegister.registerCoupon(userId, couponCode);
    }

    public List<AvailableUserCouponItem> getAvailableUserCoupons(Long userId) {
        return userCouponReader.getAvailableUserCoupons(userId);
    }
}

File: qc-core/qc-pay/qc-pay-api/src/main/kotlin/com/eager/questioncloud/payment/coupon/service/UserCouponService.kt
package com.eager.questioncloud.payment.coupon.service

import com.eager.questioncloud.payment.coupon.implement.UserCouponRegister
import com.eager.questioncloud.payment.dto.AvailableUserCoupon
import com.eager.questioncloud.payment.repository.UserCouponRepository
import org.springframework.stereotype.Service

@Service
class UserCouponService(
    private val userCouponRegister: UserCouponRegister,
    private val userCouponRepository: UserCouponRepository,
) {
    fun registerCoupon(userId: Long, couponCode: String) {
        userCouponRegister.registerCoupon(userId, couponCode)
    }
    
    fun getAvailableUserCoupons(userId: Long): List<AvailableUserCoupon> {
        return userCouponRepository.getAvailableUserCoupons(userId)
    }
}
